.github/workflows/nightly-publish.yml

```diff
@@ -23,3 +23,9 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.NIGHTLY_BUILD_GH_TOKEN }}
         run: |
           gh release create  nightly-${NIGHTLY_DATE} -R ${{github.repository}} --title "MapTool Nightly ${NIGHTLY_DATE}"  --notes "MapTool Nightly Build for ${NIGHTLY_DATE}" --generate-notes -p --target develop
+      - name: Delete old Nightly releases
+        id: delete-old
+        env:
+          GITHUB_TOKEN: ${{ secrets.NIGHTLY_BUILD_GH_TOKEN }}
+        run: |
+          gh release list -R ${{ github.repository}} | grep nightly | tail +2 |  awk '{ print $5}' | xargs -n 1 gh release delete -R ${{ github.repository}} -y --cleanup-tag
```

Comment:
An alternative for your consideration that is not dependent on `gh` output formatting:
```
gh release list ${{ github.repository}} --json 'name,tagName --jq '.[] | select(.tagName | startswith("nightly"))|.tagName' | xargs -n 1 gh release delete -R ${{ github.repository}} -y --cleanup-tag
```

Reply:
Upgdated to use the gh query functionality